packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -162,11 +162,13 @@ class _MaterialTextSelectionControls extends TextSelectionControls {
         : startTextSelectionPoint.point.dy + _kToolbarHeight + _kToolbarContentDistanceBelow;
     final Offset preciseMidpoint = Offset(position.dx, y);
 
+    // get the Overlay's render box
+    RenderBox overlayBox = Overlay.of(context).context?.findRenderObject();
```

Comment:
Why `context?`?  

If that's actually intended, then we should handle the case where this is null below - I'm not super familiar with the overlay API so I'm not sure if that's something we need to be concerned with though.

Reply:
Also, this should be `final`.

Reply:
Thanks @dnfield for the review. Yes, I was thinking about this same thing over the weekend actually. I don't think the context would be null if we can actually find it with the `.of` method, but I wasn't sure if we could always assume that. I was going to make the next line below be `overlayBox?.size ?? MediaQuery.of(context).size` but I'd rather wait to get more feedback on whether this is actually needed or not.

Reply:
I'm pretty sure you would always have the context in this method, and if not it'd be a bug.  The context is just basically the `Element` for the widget, so if you can't find your parent element you're in trouble 😆 

We could add an `assert` to that effect - e.g. 
```dart
final OverlayState overlayState = Overlay.of(context);
assert(overlayState != null);
assert(overlayState.context != null);
```